### PR TITLE
Why did the cow cross the road sol

### PR DESCRIPTION
**Pull request by HarryWangATX**
_If any of the below doesn't apply to this Pull Request, mark the checkbox as
done._

- [x] I have tested my code
- [x] I have followed the code style guidelines mentioned
      [here](https://usaco.guide/general/adding-solution/#code-conventions)
- [x] I have properly formatted the files according to the steps mentioned
      [here](https://usaco.guide/general/adding-solution#steps)

### CODE REVIEW

File: solutions/usaco-717.mdx [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+## Implementation
+
+<LanguageSection>
+<JavaSection>
+
+```java
+import java.util.*;
+import java.io.*;
+
+public class visitfj {
```
Comment (SansPapyrus683) [inline solutions/usaco-717.mdx:19]:
```suggestion
public class VisitFJ {
```

File: solutions/usaco-717.mdx [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+				int ny = pos[1] + dy[i];
+
+				if(nx < 0 || nx >= N || ny < 0 || ny >= N) continue;
+
+				int nw = cur.w + w[nx][ny] + T * 3;
+
+				if(nw > dist[nx][ny]) continue;
+
+				dist[nx][ny] = nw;
+
```
Comment (SansPapyrus683) [inline solutions/usaco-717.mdx:64]: add braces and remove all whitespace

File: solutions/usaco-717.mdx [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+		pq.add(new Edge(new int[]{0,0}, 0));
+
+		while(!pq.isEmpty()) {
+			Edge cur = pq.poll();
+			int[] pos = cur.t;
+
+			if(N-1-pos[0] + N-1-pos[1] < 3 && N-1-pos[0] + N-1-pos[1] > 0) {
+				dist[N-1][N-1] = Math.min(dist[N-1][N-1], cur.w + T * (N-1-pos[0] + N-1-pos[1]));
+			}
```
Comment (SansPapyrus683) [inline solutions/usaco-717.mdx:51]: spaces.
(also other places as well)

File: solutions/usaco-717.mdx [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+```java
+import java.util.*;
+import java.io.*;
+
+public class visitfj {
+	static int[] dx = {0, 1, 2, 3, 0, 1, 2, -1, -2, -3, -1, -2, 1, -1, 0, 0};
+	static int[] dy = {3, 2, 1, 0, -3, -2, -1, 2, 1, 0, -2, -1, 0, 0, 1, -1};
+	static int N, T;
+	static int[][] w;
+	static int[][] dist;
```
Comment (SansPapyrus683) [inline solutions/usaco-717.mdx:24]: globals not needed

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: solutions/usaco-717.mdx (hunk 4 above)
Comment (nwatx) [reply solutions/usaco-717.mdx:24]: ^

File: solutions/usaco-717.mdx [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+
+		PriorityQueue<Edge> pq = new PriorityQueue<>();
+
+		pq.add(new Edge(new int[]{0,0}, 0));
+
+		while(!pq.isEmpty()) {
+			Edge cur = pq.poll();
+			int[] pos = cur.t;
+
+			if(N-1-pos[0] + N-1-pos[1] < 3 && N-1-pos[0] + N-1-pos[1] > 0) {
```
Comment (nwatx) [inline solutions/usaco-717.mdx:49]: also, is it possible to make this more readable

**Comment by SansPapyrus683** [comment]
comments pwese uwu

**Review by nwatx — APPROVED** [review]

File: solutions/usaco-717.mdx [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+
+		for (int i = 0; i < N; i++) {
+			Arrays.fill(dist[i], Integer.MAX_VALUE);
+			for (int j = 0; j < N; j++) {
+				w[i][j] = io.nextInt();
+			}
+		}
+
+		PriorityQueue<Edge> pq = new PriorityQueue<>();
+		pq.add(new Edge(new int[]{0,0}, 0));
```
Comment (nwatx) [inline solutions/usaco-717.mdx:40]:
```suggestion
		pq.add(new Edge(new int[]{0, 0}, 0));
```

File: solutions/usaco-717.mdx [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+		while (!pq.isEmpty()) {
+			Edge cur = pq.poll();
+			int[] pos = cur.t;
+
+			int d = N - 1 - pos[0] + N - 1 - pos[1];
+
+			if (d < 3 && d > 0) {
+				dist[N-1][N-1] = Math.min(dist[N-1][N-1], cur.w + T * d);
```
Comment (nwatx) [inline solutions/usaco-717.mdx:48]: operator spacing

**Review by nwatx — CHANGES_REQUESTED** [review]

**Review by dongliu0426 — APPROVED** [review]

**Review by nwatx — APPROVED** [review]

**Pull request merged**